SIPs/sip_001.md

```diff
@@ -0,0 +1,172 @@
+# SPOOL Improvement Proposal 001
+
+
+## Abstract
+
+This proposal discusses details for allowing the implementation
+of unlimited editions in the SPOOL ownership API.
+
+## Requirements
+
+    - An API user must be able to bulk create a set of unlimited editions for a piece in a single
+      HTTP request
+    - An API user must be able to repeatedly create a set of unlimited editions
+      for a piece using HTTP requests
+    - An API user is not required to know about concurrent edition creation
+      processes running on a piece they try to create editions on
+      (meaning an API user can submit a request for creating editions, and even
+      though this process is still running, can send another request for
+      creating more editions)
+    - An API user must be able to create a distinct set of editions which
+      numbers do neither need to be in order, nor do they need to be consecutive
+      (All these examples shall be valid: [1, 2, 3], [3, 1, 2], [3, 1])
+    - An API user must be able to `freeze` or limit the unlimited editions of a
+      piece using a single HTTP request
+    - An API user must be able to track the creation of editions of a piece in a
+      modern-web kind of fashion (e.g. using websockets, instead of polling)
```

Comment:
Given that [websocket](https://en.wikipedia.org/wiki/WebSocket) is a documented protocol standardized by the IETF as RFC 6455, could we phrase this more formally, in such a way that other protocols can be added or proposed. 

Also, and perhaps more importantly, the API consumers may differ, and may not be frontend/browser related. Hence, perhaps we could simply say that the API shoudl be capable to support a pub-sub mechanism? The API being the publisher of messages/notifications, to which clients can subscribe via different means.